components/Portfolio/AssetDisplay/AssetDisplay.tsx

```diff
@@ -1,36 +1,221 @@
+  assetCurrent: AssetCurrent | undefined;
+  assetHistory: AssetHistory | undefined;
 };
 
-const AssetDisplay = ({ asset, currentResponse, historyResponse }: Props) => {
-  const { coinName, date } = asset;
+const AssetDisplay = ({ asset, assetCurrent, assetHistory }: Props) => {
+  const { coinName, coinSymbol, coinImage, date } = asset;
+  const currency = useUserCurrencySetting();
+
+  const editButtonId = useId();
+  const displayDate = extractDate(date).toLocaleString("en-US", {
+    dateStyle: "medium",
+  });
+
+  const maybeDisplayData = assetDisplayData(
+    asset,
+    currency,
+    assetCurrent,
+    assetHistory
+  );
+
+  const placeholder = <span className="animate-pulse">--</span>;
+
+  const currentValue = (() => {
+    const value = maybeDisplayData?.currentValue;
+    if (!value) return placeholder;
+    return <span>{currencyMap.get(currency) + localeFormat(value)}</span>;
+  })();
+
+  const currentValueChangePercent = (() => {
+    const change = maybeDisplayData?.currentValueChangePercent;
+    if (!change) return "";
+    return (
+      <span className={cn(change > 0 ? "text-market-up" : "text-market-down")}>
+        ( {change > 0 && "+"}
+        {localeFormat(change)}% )
+      </span>
+    );
+  })();
+
+  const currentPrice = (() => {
+    const price = maybeDisplayData?.currentPrice;
+    if (!price) return placeholder;
+    if (price < 0.01) {
+      return <span>{currencyMap.get(currency) + price.toExponential(2)}</span>;
+    } else
+      return <span>{currencyMap.get(currency) + localeFormat(price)}</span>;
+  })();
+
+  const currentPriceChange24h = (() => {
+    const change = maybeDisplayData?.currentPriceChange24h;
+    if (!change) return placeholder;
+    return (
+      <>
+        <span>
+          <CaretIcon
+            className={cn(
+              "w-4 h-4 mr-1 inline",
+              change > 0 ? "fill-market-up" : "fill-market-down rotate-180"
+            )}
+          />
+        </span>
+        <span
+          className={cn(change > 0 ? "text-market-up" : "text-market-down")}
+        >
+          {localeFormat(Math.abs(change))}%
+        </span>
+      </>
+    );
+  })();
+
+  const marketCapVsVolume = (() => {
+    const ratio = maybeDisplayData?.marketCapVsVolume;
+    const change = maybeDisplayData?.currentPriceChange24h;
+    if (!ratio || !change) return placeholder;
+    return (
+      <>
+        <span
+          className={cn(change > 0 ? "text-market-up" : "text-market-down")}
+        >
+          {Math.round(ratio)}%
+        </span>
+        <ProgressWidget
+          containerClassName={cn(
+            "w-full ml-4 mr-2",
+            change > 0 ? "bg-market-up/20" : "bg-market-down/20"
+          )}
+          progressClassName={cn(
+            "bg-market-down",
+            change > 0 ? "bg-market-up" : "bg-market-down"
+          )}
+          progressPercentage={ratio}
+        />
+      </>
+    );
+  })();
+
+  const circVsTotalSupply = (() => {
```

Comment:
Not to sacrifice code readability, I wouldn't change the if statement here. Once a ternary is used, it might simplify the logic but make it a bit harder to understand the code. It's always a balance between code simplicity and readability.